lib/system/extra/cmd_streamer.ex

```diff
@@ -0,0 +1,143 @@
+defmodule System.Extra.CmdStreamer do
+  @moduledoc """
+  Module allowing Enumerables to be constructed around Shell commands.
+
+  """
+
+  alias __MODULE__
+
+
+
+  #########################################################
+  # Types
+  #########################################################
+
+  @opaque t :: %__MODULE__{
+    port: port,
+    queue: :queue.queue,
+    receiver: pid,
+    running: boolean,
+    pending_demand: non_neg_integer,
+  }
+
+  defstruct [
+    port: nil,
+    queue: :queue.new(),
+    receiver: nil,
+    running: true,
+    pending_demand: 0,
+  ]
+
+
+
+  #########################################################
+  # Public API
+  #########################################################
+
+  @doc """
+  Starts the `CmdStreamer` enumerable by wrapping `cmd` and forwarding `args` to it.
+
+  """
+  @spec start(binary, [binary]) :: pid
+  def start(cmd, args) do
+    receiver =
+      self()
+
+    cmd_string =
+      [cmd | args] |> Enum.join(" ")
```

Comment:
Oh man be very careful with this. If there are spaces in one of the arguments then this would have worked with other exec calls, but will break here. We should look into using `{:spawn_executable, filename}` with the `:args` options.